system/jlib/jevent.cpp

```diff
@@ -510,22 +510,23 @@ void EventRecorder::recordRecordingActive(bool enabled)
     writeEventFooter(pos, requiredSize, writeOffset);
 }
 
-void EventRecorder::recordIndexLookup(unsigned fileid, offset_t offset, byte nodeKind, bool hit)
+void EventRecorder::recordIndexLookup(unsigned fileid, offset_t offset, byte nodeKind, bool hit, size32_t sizeIfHit)
 {
     if (!isRecording())
         return;
 
     if (unlikely(outputToLog))
-        TRACEEVENT("{ \"name\": \"IndexLookup\", \"file\": %u, \"offset\"=0x%llx, \"kind\": %d, \"hit\": %s }", fileid, offset, nodeKind, boolToStr(hit));
+        TRACEEVENT("{ \"name\": \"IndexLookup\", \"file\": %u, \"offset\"=0x%llx, \"kind\": %d, \"hit\": %s, \"size\": %u }", fileid, offset, nodeKind, boolToStr(hit), sizeIfHit);
 
-    size32_t requiredSize = sizeMessageHeaderFooter + getSizeOfAttrs(fileid, offset, nodeKind, hit);
+    size32_t requiredSize = sizeMessageHeaderFooter + getSizeOfAttrs(fileid, offset, nodeKind, hit, sizeIfHit);
```

Comment:
I did consider only adding it if hit is true.  What do you think?

Reply:
I suppose if it's a value I care about in a visitor, I would default a variable receiving it to zero. I can't think of a reason I would need an explicit visit to affirm that. At this time I would include it conditionally, but I reserve the right to complain if something changes.